src/openms/include/OpenMS/METADATA/ID/QueryMatchGroup.h

```diff
@@ -62,7 +62,7 @@ namespace OpenMS
         for (auto it = ++query_match_refs.begin(); it != query_match_refs.end();
              ++it)
         {
-          if ((*it)->identified_molecule_ref != ref) return false;
+          if (!((*it)->identified_molecule_ref == ref)) return false;
```

Comment:
looks like 5b72555e16e845c4db8a3fff09f2160e020c7cfc (almost) fixed the pyOpenMS build